Fix #519 Prevent infinite loop when getting references from invalid range

rdarcy1 opened the pull request:
Fixes #519 

Didn't realise it was such an easy fix..! Currently it returns an empty array of references when passed an invalid range, not sure if an exception would be more suitable.

rdarcy1 commented:
I've updated to throw an exception on invalid range and amended the unit test to match.

criztovyl commented:
Try executing `./vendor/bin/php-cs-fixer fix` to fix the remaining code-style issues that Travis does not like.

Regarding Scrutinizer I am not sure, but last time PowerKiki suggested to extract the added code into it's own method, maybe try that. :)

Thx!

criztovyl commented:
In this case just your added code in `extractAllCellReferencesInRange()` would need to be extracted.

rdarcy1 commented:
I've also refactored the large `extractAllCellReferencesInRange` method down to a couple of smaller methods and extracted a couple of duplicated lines to improve the code rating – hopefully that's not outside the scope of the original PR?

I think much of the logic concerning cell ranges and references could be extracted to their own objects rather than dealing with range and reference arrays if you were open to cleaning up the class further.

criztovyl commented:
By personal preference I would have only extracted as much as is required to pass CI and included everything else in a additional PR. :)
But that seems to be exactly what you did ^^

Could provide an additional PR for your suggested refactorings? Otherwise you could open an a feature request issue.

Thanks for your contributions so far. :)

PowerKiKi commented:
Thanks ! The refactoring is always welcomed. The existing code tends to use way too big methods...

boredom2 commented:
Hi there,
we currently experience Tons of Problems with this Change :)
After a quick Investigation, I suppose, that $startCol>=$endCol wrongly fails in the case "B" >= "AG".
Obviously, "B" is smaller than "AG", but due to PHPs String Comparison, it returns false and therefore wrongly emits the Exception.

rdarcy1 commented:
Ah, sorry I wasn't thorough enough with my tests then. I can re-submit with a better comparison function.

PowerKiKi commented:
@boredom2, I can confirm what you say. It was merged too quickly. It was also reported in #545, and I just pushed a fix on develop branch. Can you confirm it fixes the issue for you ?

PowerKiKi commented:
It's been released as [1.3.1](https://github.com/PHPOffice/PhpSpreadsheet/releases/tag/1.3.1)
